crash/app/src/main/java/com/google/samples/quickstart/crash/kotlin/CustomKeySamples.kt

```diff
@@ -0,0 +1,175 @@
+package com.google.samples.quickstart.crash.kotlin
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.pm.PackageManager
+import android.net.ConnectivityManager
+import android.net.ConnectivityManager.NetworkCallback
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.os.Build
+import android.telephony.TelephonyManager
+import android.view.View
+import androidx.core.content.ContextCompat
+import com.google.android.gms.common.GoogleApiAvailability
+import com.google.firebase.crashlytics.CustomKeysAndValues
+import com.google.firebase.crashlytics.FirebaseCrashlytics
+
+/**
+ * The following are samples of custom keys that may be useful to record via Crashlytics prior to a Crash.
+ *
+ * These utility methods are not meant to be comprehensive, but they are illustrative of the types of things you
+ * could track using custom keys in Crashlytics.
+ */
+class CustomKeySamples(private val context: Context) {
+
+    /**
+     * Set a subset of custom keys simultaneously.
+     */
+    fun setSampleCustomKeys() {
+        FirebaseCrashlytics.getInstance().setCustomKeys(CustomKeysAndValues.Builder()
+                .putString("Locale", locale)
+                .putFloat("Screen Density", density)
+                .putString("Google Play Services Availability", googlePlayServicesAvailability)
+                .putString("Os Version", osVersion)
+                .putString("Install Source", installSource)
+                .putString("ABIs", preferredAbi).build())
+    }
+
+    /**
+     * Update network state and add a hook to update network state going forward.
+     */
+    fun updateAndTrackNetworkState() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+            connectivityManager
+                    .getNetworkCapabilities(connectivityManager.activeNetwork)
+                    ?.let { updateNetworkCapabilityCustomKeys(it) }
+
+            // Set up a callback to match our best-practices around custom keys being up-to-date
+            connectivityManager.registerDefaultNetworkCallback(object : NetworkCallback() {
+                override fun onCapabilitiesChanged(network: Network, networkCapabilities: NetworkCapabilities) {
+                    updateNetworkCapabilityCustomKeys(networkCapabilities)
+                }
+            })
+        }
+    }
+
+    private fun updateNetworkCapabilityCustomKeys(networkCapabilities: NetworkCapabilities) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            val instance = FirebaseCrashlytics.getInstance()
```

Comment:
For the Kotlin samples, we prefer to use Firebase KTX 
```suggestion
            val instance = Firebase.crashlytics
```

This applies throughout.

Reply:
updated